File: jason/asSyntax/Term.java
package jason.asSyntax;

import jason.asSemantics.Unifier;
import jason.util.ToDOM;

import java.io.Serializable;
import java.util.Map;

/**
 * Common interface for all kind of terms
 */
public interface Term extends Cloneable, Comparable<Term>, Serializable, ToDOM {

    public boolean isVar();
    public boolean isUnnamedVar();
    public boolean isLiteral();
    public boolean isRule();
    public boolean isList();
    public boolean isString();
    public boolean isInternalAction();
    public boolean isArithExpr();
    public boolean isNumeric();
    public boolean isPred();
    public boolean isGround();
    public boolean isStructure();
    public boolean isAtom();
    public boolean isPlanBody();

    public boolean hasVar(VarTerm t);
    public void countVars(Map<VarTerm, Integer> c);

    public Term clone();

    public boolean equals(Object o);
    
    /** 
     *  Replaces variables by their values in the unifier, returns true
     *  if some variable was applied.  
     */
    public boolean apply(Unifier u);

    /** Removes the value cached for hashCode */
    public void resetHashCodeCache();
    
    public int getSrcLine();
    public String getSrc();
    public String getErrorMsg(); // info for error messages 

}

File: dotNetSrc/Jason.NET/Jason.asSyntax/Term.cs
using System;
using System.Runtime.Serialization;
using Jason.Utils;

namespace Jason.asSyntax
{
	public interface Term : ICloneable, IComparable<Term>, ISerializable, ToDOM {

		 bool isVar();
		 bool isUnnamedVar();
	 	 bool isLiteral();
		 bool isRule();
		 bool isList();
		 bool isString();
		 bool isInternalAction();
		 bool isArithExpr();
		 bool isNumeric();
		 bool isPred();
		 bool isGround();
		 bool isStructure();
		 bool isAtom();
		 bool isPlanBody();

		 /// TODO : add VarTerm 
//		 bool hasVar(VarTerm t);
//		 void countVars(Map<VarTerm, Integer> c);

		 Term clone();

		 bool equals(Object o);
		
		/** replaces variables by their values in the unifier, returns true if some variable was applied */
		/// TODO : add Unifier 
//		 bool apply(Unifier u);

		 void setSrcInfo(SourceInfo s);
		 SourceInfo getSrcInfo();
	}

}
